luminoth/utils/image_test.py

```diff
@@ -261,6 +319,126 @@ def testFlipBboxesDiffDtype(self):
+
+        When bboxes is None.
+        """
+        im_shape = (600, 800, 3)
+        image = self._gen_image(*im_shape)
+        config = self._random_patch_config
+        ret_image, ret_bboxes = self._random_patch(image, config)
+        # Assertions
+        self.assertTrue(np.all(ret_image.shape <= im_shape))
+        # We ran return_dict.get('bboxes') on the dict returned by
+        # random_patch. That's why we should get a None in this case.
+        self.assertIs(ret_bboxes, None)
+
+    def testRandomResizeImageBboxes(self):
+        """Tests the integrity of the return values of random_resize
+
+        This tests the case when bboxes is not None.
+        """
+        im_shape = (600, 800, 3)
+        config = self._random_resize_config
+        total_boxes = 35
+        label = 3
+
+        image, bboxes = self._get_image_with_boxes(im_shape, total_boxes)
+        # Add a label to each bbox.
+        bboxes_w_label = tf.concat(
+            [
+                bboxes,
+                tf.fill((bboxes.shape[0], 1), label)
+            ],
+            axis=1
+        )
+        ret_image, ret_bboxes = self._random_resize(
+            image, config, bboxes_w_label
+        )
+        # Assertions
+        self.assertEqual(ret_bboxes.shape[0], total_boxes)
+        self.assertTrue(np.all(
+            np.asarray(ret_image.shape[:2]) >= config.min_size
+        ))
+        self.assertTrue(np.all(
+            np.asarray(ret_image.shape[:2]) <= config.max_size
+        ))
+
+    def testRandomResizeOnlyImage(self):
+        """Tests the integrity of the return values of random_resize
+
+        This tests the case when bboxes is None.
+        """
+        im_shape = (600, 800, 3)
+        image = self._gen_image(*im_shape)
+        config = self._random_resize_config
+        ret_image, ret_bboxes = self._random_resize(image, config)
+        # Assertions
+        self.assertEqual(ret_bboxes, None)
+        self.assertTrue(np.all(
+            np.asarray(ret_image.shape[:2]) >= config.min_size
+        ))
+        self.assertTrue(np.all(
+            np.asarray(ret_image.shape[:2]) <= config.max_size
+        ))
+
+    def testRandomDistort(self):
+        """Tests the integrity of the return values of random_distortion.
+        """
+        im_shape = (600, 900, 3)
+        config = self._random_distort_config
+        total_boxes = 5
+        label = 3
+
+        image, bboxes = self._get_image_with_boxes(im_shape, total_boxes)
+        # Add a label to each bbox.
+        bboxes_w_label = tf.concat(
+            [
+                bboxes,
+                tf.fill((bboxes.shape[0], 1), label)
+            ],
+            axis=1
+        )
+
+        ret_image, ret_bboxes = self._random_distort(
+            image, config, bboxes_w_label
+        )
+        # Assertions
+        self.assertEqual(im_shape, ret_image.shape)
+        self.assertAllClose(
+            bboxes, ret_bboxes[:, :4], atol=self._equality_delta
+        )
+
```

Comment:
Add assert with small distortions in which the resulting image is allClose within a rtol of ~5% of the original image.